client/lib/analytics/index.js

```diff
@@ -271,12 +280,20 @@ const analytics = {
 				for ( const key in eventProperties ) {
 					if ( isObjectLike( eventProperties[ key ] ) && typeof console !== 'undefined' ) {
 						const errorMessage =
-							`Unable to record event "${ eventName }" because nested` +
+							`Tracks: Unable to record event "${ eventName }" because nested` +
 							`properties are not supported by Tracks. Check '${ key }' on`;
 						console.error( errorMessage, eventProperties ); //eslint-disable-line no-console
 
 						return;
 					}
+					if ( key !== snakeCase( key ) ) {
```

Comment:
`snakeCase` is an improvement, but we could bring over the backend regex: `^[a-z_][a-z0-9_]*$`

Reply:
done! thanks.